pkg/controller/core/clusterqueue_controller.go

```diff
@@ -107,18 +109,38 @@ func (r *ClusterQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		}
 	}
 
-	status, err := r.Status(&cqObj)
-	if err != nil {
-		log.Error(err, "Failed getting status from cache")
-		return ctrl.Result{}, err
-	}
-
-	if !equality.Semantic.DeepEqual(status, cqObj.Status) {
-		cqObj.Status = status
-		err := r.client.Status().Update(ctx, &cqObj)
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+	newCQObj := cqObj.DeepCopy()
+	if r.cache.ClusterQueueActive(cqObj.Name) {
+		msg := fmt.Sprintf("ClusterQueue %q can admit new workloads", cqObj.Name)
+		reason := "Ready"
+
+		// ClusterQueueActive=True
+		updatedActiveCondition := r.UpdateClusterQueueCondition(&newCQObj.Status,
+			kueue.ClusterQueueActive, corev1.ConditionTrue, reason, msg)
+		// ClusterQueuePending=False
+		updatedPendingCondition := r.UpdateClusterQueueCondition(&newCQObj.Status,
+			kueue.ClusterQueuePending, corev1.ConditionFalse, reason, msg)
+
+		updatedConditions := updatedActiveCondition || updatedPendingCondition
+		if err := r.UpdateClusterQueueStatus(ctx, newCQObj, updatedConditions); err != nil {
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		}
+	} else if r.cache.ClusterQueuePending(cqObj.Name) {
+		msg := fmt.Sprintf("ClusterQueue %q can't admit new workloads", cqObj.Name)
```

Comment:
It would be useful to know why: For example: there are missing ResourceFlavors or it's being deleted

Reply:
That makes sense.

Reply:
Done.